pkg/client/custom_api/client.go

```diff
@@ -79,12 +79,12 @@ func (mc *metricsClient) ListCustomMetricInfos() (map[provider.CustomMetricInfo]
 	metricInfos := make(map[provider.CustomMetricInfo]struct{})
 	namer, err := config.NewNamer(mc.metricServerCfg.Rename)
 	if err != nil {
-		klog.Fatalf("%s: failed to create customMetricNamer: %v", mc.GetConfiguration().Name, err)
+		return nil, fmt.Errorf("%s: failed to create customMetricNamer: %v", mc.GetConfiguration().Name, err)
```

Comment:
this is a behaviour change IIUC previously the program would exit, but maybe that is OK, thoughts?

Reply:
Yes, I changed it. It seems wrong to me to restart the pod for that. This is now an error that will be logged by the refreshMetrics caller.